scripts/on-prem-archive.sh

```diff
@@ -95,7 +103,7 @@ cleanup() {
 }
 
 download_latest_archive() {
-  curl -O "$s3_root_url/$marker"
+  hab pkg exec core/curl curl -O "$s3_root_url/$marker"
```

Comment:
We may also want to pull out the hab pkg execs into variables so that we can say something like ${CURL_CMD} instead of the hab pkg exec, and its also easier to genericize